File: mobile-engage/src/main/java/com/emarsys/mobileengage/iam/model/buttonclicked/ButtonClickedRepository.java
package com.emarsys.mobileengage.iam.model.buttonclicked;

import android.content.ContentValues;
import android.database.Cursor;

import com.emarsys.core.database.helper.DbHelper;
import com.emarsys.core.database.repository.AbstractSqliteRepository;

import static com.emarsys.core.database.DatabaseContract.BUTTON_CLICKED_COLUMN_NAME_BUTTON_ID;
import static com.emarsys.core.database.DatabaseContract.BUTTON_CLICKED_COLUMN_NAME_CAMPAIGN_ID;
import static com.emarsys.core.database.DatabaseContract.BUTTON_CLICKED_COLUMN_NAME_TIMESTAMP;
import static com.emarsys.core.database.DatabaseContract.BUTTON_CLICKED_TABLE_NAME;


public class ButtonClickedRepository extends AbstractSqliteRepository<ButtonClicked> {

    public ButtonClickedRepository(DbHelper dbHelper) {
        super(BUTTON_CLICKED_TABLE_NAME, dbHelper);
    }

    @Override
    protected ContentValues contentValuesFromItem(ButtonClicked item) {
        ContentValues contentValues = new ContentValues();
        contentValues.put(BUTTON_CLICKED_COLUMN_NAME_CAMPAIGN_ID, item.getCampaignId());
        contentValues.put(BUTTON_CLICKED_COLUMN_NAME_BUTTON_ID, item.getButtonId());
        contentValues.put(BUTTON_CLICKED_COLUMN_NAME_TIMESTAMP, item.getTimestamp());
        return contentValues;
    }

    @Override
    protected ButtonClicked itemFromCursor(Cursor cursor) {
        String campaignId = cursor.getString(cursor.getColumnIndex(BUTTON_CLICKED_COLUMN_NAME_CAMPAIGN_ID));
        String buttonId = cursor.getString(cursor.getColumnIndex(BUTTON_CLICKED_COLUMN_NAME_BUTTON_ID));
        long timestamp = cursor.getLong(cursor.getColumnIndex(BUTTON_CLICKED_COLUMN_NAME_TIMESTAMP));
        return new ButtonClicked(campaignId, buttonId, timestamp);
    }

}

File: mobile-engage/src/main/java/com/emarsys/mobileengage/iam/model/buttonclicked/ButtonClickedRepository.kt
package com.emarsys.mobileengage.iam.model.buttonclicked

import android.content.ContentValues
import android.database.Cursor
import com.emarsys.core.Mockable
import com.emarsys.core.database.DatabaseContract
import com.emarsys.core.database.helper.DbHelper
import com.emarsys.core.database.repository.AbstractSqliteRepository
import com.emarsys.core.handler.ConcurrentHandlerHolder

@Mockable
class ButtonClickedRepository(
    dbHelper: DbHelper,
    concurrentHandlerHolder: ConcurrentHandlerHolder
) : AbstractSqliteRepository<ButtonClicked>(
    DatabaseContract.BUTTON_CLICKED_TABLE_NAME,
    dbHelper,
    concurrentHandlerHolder
) {
    override fun contentValuesFromItem(item: ButtonClicked): ContentValues {
        val contentValues = ContentValues()
        contentValues.put(DatabaseContract.BUTTON_CLICKED_COLUMN_NAME_CAMPAIGN_ID, item.campaignId)
        contentValues.put(DatabaseContract.BUTTON_CLICKED_COLUMN_NAME_BUTTON_ID, item.buttonId)
        contentValues.put(DatabaseContract.BUTTON_CLICKED_COLUMN_NAME_TIMESTAMP, item.timestamp)
        return contentValues
    }

    override fun itemFromCursor(cursor: Cursor): ButtonClicked {
        val campaignId =
            cursor.getString(cursor.getColumnIndexOrThrow(DatabaseContract.BUTTON_CLICKED_COLUMN_NAME_CAMPAIGN_ID))
        val buttonId = cursor.getString(cursor.getColumnIndexOrThrow(DatabaseContract.BUTTON_CLICKED_COLUMN_NAME_BUTTON_ID))
        val timestamp = cursor.getLong(cursor.getColumnIndexOrThrow(DatabaseContract.BUTTON_CLICKED_COLUMN_NAME_TIMESTAMP))
        return ButtonClicked(campaignId, buttonId, timestamp)
    }
}